src/data/polynomial/module.lean

```diff
@@ -0,0 +1,143 @@
+/-
+Copyright (c) 2022 Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang
+-/
+
+import ring_theory.finiteness
+
+/-!
+# Polynomial module
+
+In this file, we define the polynomial module for an `R`-module `M`, i.e. the `R[X]`-module `M[X]`.
+
+This is defined as an type alias `polynomial_module M := ℕ →₀ M`, since there might be different
+module structures on `ℕ →₀ M` of interest.
```

Comment:
Does it make sense to use `polynomial M` (I don't know)?

Reply:
It clashes with `_root_.polynomial` for polynomial rings.

Reply:
I meant reusing `_root_.polynomial`. Do these two types have some conflicting instances (sorry for asking stupid questions)?

Reply:
`_root_.polynomial R` is based on `add_monoid_algebra`, and requires `R` to have a `semiring` structure. I'm not sure if we should change the definition of `_root_.polynomial` altogether?